Userland/Applications/MouseSettings/SchemeWidget.cpp

```diff
@@ -0,0 +1,138 @@
+
+String MouseCursorModel::column_name(int column_index) const
+{
+    switch (column_index) {
+    case Column::Bitmap:
+        return {};
+    case Column::Name:
+        return "Name";
+    }
+    VERIFY_NOT_REACHED();
+}
+
+GUI::Variant MouseCursorModel::data(GUI::ModelIndex const& index, GUI::ModelRole role) const
+{
+    auto& cursor = m_cursors[index.row()];
+
+    if (role == GUI::ModelRole::Display) {
+        switch (index.column()) {
+        case Column::Bitmap:
+            if (!cursor.bitmap)
+                return {};
+            return *cursor.bitmap;
+        case Column::Name:
+            return cursor.name;
+        }
+        VERIFY_NOT_REACHED();
+    }
+    return {};
+}
+
+void MouseCursorModel::update()
+{
+    if (m_scheme_name.is_empty())
+        return;
+
+    m_cursors.clear();
+    Core::DirIterator iterator(String::formatted("/res/cursor-schemes/{}", m_scheme_name), Core::DirIterator::Flags::SkipDots);
+
+    while (iterator.has_next()) {
+        auto path = iterator.next_full_path();
+        if (path.ends_with(".ini"))
+            continue;
+        if (path.contains("2x"))
+            continue;
+        Cursor cursor;
+        cursor.path = move(path);
+        cursor.bitmap = Gfx::Bitmap::try_load_from_file(cursor.path);
+        auto filename_split = cursor.path.split('/');
+        cursor.name = filename_split[3];
+        m_cursors.append(move(cursor));
+    }
+
+    did_update();
+}
+
+GUI::Variant SchemeModel::data(GUI::ModelIndex const& index, GUI::ModelRole role) const
+{
+    if (role == GUI::ModelRole::Display) {
+        return m_schemes[index.row()];
+    }
+    return {};
+}
+
+void SchemeModel::update()
+{
+    m_schemes.clear();
+
+    Core::DirIterator iterator("/res/cursor-schemes", Core::DirIterator::Flags::SkipDots);
+
+    while (iterator.has_next()) {
+        auto path = iterator.next_path();
+        if (access(String::formatted("/res/cursor-schemes/{}/Config.ini", path).characters(), R_OK) == 0)
+            m_schemes.append(path);
+    }
+
+    did_update();
+}
+
+SchemeWidget::SchemeWidget()
+{
+    load_from_gml(scheme_widget_gml);
+    m_cursors_tableview = find_descendant_of_type_named<GUI::TableView>("cursors_tableview");
+    m_cursors_tableview->set_highlight_selected_rows(true);
+    m_cursors_tableview->set_alternating_row_colors(false);
+    m_cursors_tableview->set_vertical_padding(16);
+    m_cursors_tableview->set_column_headers_visible(false);
+    m_cursors_tableview->set_highlight_key_column(false);
+
+    auto mouse_cursor_model = MouseCursorModel::create();
+    auto sorting_proxy_model = GUI::SortingProxyModel::create(mouse_cursor_model);
+    sorting_proxy_model->set_sort_role(GUI::ModelRole::Display);
+
+    m_cursors_tableview->set_model(sorting_proxy_model);
+    m_cursors_tableview->set_key_column_and_sort_order(MouseCursorModel::Column::Name, GUI::SortOrder::Ascending);
+    m_cursors_tableview->set_column_width(0, 25);
```

Comment:
This doesn't work for some reason:
![image](https://user-images.githubusercontent.com/42967349/127838575-dd4c39a6-7951-4918-a5a4-da971718a728.png)

Reply:
The wait cursor is expanding the table width. Animated cursors are stored as a single .png file with all the frames, so you will need to perform some modification before being able to use the animated icon bitmaps. Would be cool to have it animated in the table view though. :^)

Reply:
This works in WidgetGallery. I'll just parse the name as WindowServer does.

Reply:
I fixed that.